src/blenderbim/blenderbim/bim/module/aggregate/operator.py

```diff
@@ -290,9 +372,26 @@ def _execute(self, context):
         for obj in context.selected_objects:
             obj.select_set(False)
             
-        for rel in linked_aggregate_group[0].IsGroupedBy or []:
-            for element in rel.RelatedObjects:
-                obj = tool.Ifc.get_object(element)
-                obj.select_set(True)
+
+        group_rel = linked_aggregate_group[0].IsGroupedBy or []
+        for groupLink in group_rel:
+            parts = list(groupLink.RelatedObjects)
+            if self.select_all:
+                parts_objs = []
+                for part in parts:
+                    if part.IsDecomposedBy:
+                        for subpart in part.IsDecomposedBy[0].RelatedObjects:
+                            parts.append(subpart)
+                    parts_objs.append(part)
+
+                for element in parts_objs:
+                    obj = tool.Ifc.get_object(element)
+                    if obj:
+                        obj.select_set(True)
+            else:
+                for element in parts:
+                    obj = tool.Ifc.get_object(element)
+                    obj.select_set(True)
```

Comment:
- I suggest to use underscore, because it's a Python convention. Use `group_link`
- I would change some other variable names. What you called `parts` is actually a list of aggregates that belong to the same group, right? So it should be called "linked_aggregates". And what you called `subpart` is actually a `part`.
- Also, there is no need to create a list, for the parts, you can select them in the end of loop.

So it would become something like this:
```python
        for group_link in group_rel:
            linked_aggregates = list(group_link.RelatedObjects)
            if self.select_all:
                for aggregate in linked_aggregates:
                    if aggregate.IsDecomposedBy:
                        for part in aggregate.IsDecomposedBy[0].RelatedObjects:
                            print(part)
                            obj = tool.Ifc.get_object(part)
                            if obj:
                                obj.select_set(True)
            else:
                for aggregate in linked_aggregates:
                    obj = tool.Ifc.get_object(aggregate)
                    obj.select_set(True)

```